packages-exp/auth-exp/src/model/user.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface UserInfo {
+  readonly uid: string;
+}
+
+export interface UserParameters {
+  uid: string;
+}
+
+export class User implements UserInfo {
```

Comment:
In the prototype we had some actual code in the model/ folder and it got kinda hairy due to circular references. I think it would be best to have strictly only interfaces in model/ and have impl classes in core/

So you'd have `interface User {}` in model/ and then in core/ `class UserImpl implements User`. The only place `UserImpl` is referenced is the place where it's instantiated. Everywhere else uses the interface from model/. This way there's no circular references.

Reply:
yeah I like that, let's try it and see how far it gets us